src/key.js

```diff
@@ -728,6 +786,65 @@ User.prototype.isValidSelfCertificate = function(primaryKey, selfCertificate) {
 };
 
 /**
+ * Signs user
+ * @param  {module:packet/secret_key|module:packet/public_key} primaryKey The primary key packet
+ * @param  {Array<module:key~Key>} privateKeys decrypted private keys for signing
+ * @return {module:key~Key} new user with new certificate signatures
+ */
+User.prototype.sign = function(primaryKey, privateKeys) {
+  var user, dataToSign, signingKeyPacket, signaturePacket;
+  dataToSign = {};
+  dataToSign.key = primaryKey;
+  dataToSign.userid = this.userId || this.userAttribute;
+  user = new User(this.userId || this.userAttribute);
+  user.otherCertifications = [];
+  privateKeys.forEach(privateKey => {
+    if (privateKey.isPublic()) {
+      throw new Error('Need private key for signing');
+    }
+    if (privateKey.primaryKey.getKeyId().equals(primaryKey.getKeyId())) {
+      throw new Error('Not implemented for  self signing');
```

Comment:
Extra space will be deleted.
I wanted to prevent using this function for self signing. Because it creates more simplified signature (without preferred algorithms, key servers). Second reason is that every key-user are already signed by its key on generation stage and I don't know what happens if we will add the extra one.
Yeah, I think we can change it on comparing of fingerprints.